Formula/c/carapace.rb

```diff
@@ -0,0 +1,28 @@
+class Carapace < Formula
+  desc "Multi-shell multi-command argument completer"
+  homepage "https://carapace.sh"
+  url "https://github.com/carapace-sh/carapace-bin/archive/refs/tags/v1.0.7.tar.gz"
+  sha256 "bfcd950178023909b0854bc6bded5b57d1c00123943e9dbcdfddf2a632c71ff4"
+  license "MIT"
+  head "https://github.com/carapace-sh/carapace-bin.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "generate", "./..."
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "-tags", "release", "./cmd/carapace"
+  end
+
+  def caveats
+    <<~EOS
+            This registers all the available completers. It is also possible to load a single one by replacing _carapace with the completer name:
+              export CARAPACE_BRIDGES='zsh,fish,bash,inshellisense'
+      source <(carapace _carapace)
+    EOS
+  end
```

Comment:
The indentation needs to be fixed, but before that: is this specific to Homebrew packaging? Caveats should typically be used for that only: https://docs.brew.sh/Formula-Cookbook#caveats.

Reply:
By "specific to Homebrew packaging" I mean: do the caveats only apply to Homebrew-installed carapace? Is that not the case for upstream-built versions?

If this is not specific to Homebrew in that regard, this should be in upstream documentation rather than here.

Reply:
I see, but the hints is nessessary for users. Similar caveats blocks also exist in other formulae, such as `micromamba.rb`.

Reply:
I see, but the hints is nessessary for users. Similar caveats blocks also exist in other formulae, such as `micromamba.rb`.

Reply:
The [`micromamba` caveats](https://github.com/Homebrew/homebrew-core/blob/22a56ef0b78176baa4c5c824a4a8f5fb29f1bb39/Formula/m/micromamba.rb#L61-L67), as I understand them, are there because the user does not need to perform that initialisation task when using [the official installation method](https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html#automatic-install). However, this does not seem to be the case for carapace, as [such an initial setup is needed regardless of installation method](https://carapace-sh.github.io/carapace-bin/setup.html).

Reply:
Alright, let me remove the caveats block